### PR TITLE
Windows: continue #810

### PR DESCRIPTION
**Pull request by justinmk**
rebase/continue #810

Note: missing https://github.com/neovim/neovim/pull/810/commits/411466e2473136481f51c083fb10d1aa054b0d0d

### CODE REVIEW

**Comment by equalsraf** [comment]
Hi @justinmk 

I'm still catching up FWIW I had a couple of these things already worked out as separate branches in my own repo, probably easier to merge than all at  once (windows-clipboard, windows-terminal and a couple others).

- We can probably drop the cross compile bits, I am certainly not using them.
- I believe :terminal is broken on a libuv assert, unless this has changed since October.

i.e. clipboard, shellescape and the msvc are the last useful bits from #810, most others need to be reworked anyway.

**Comment by justinmk** [comment]
@equalsraf Awesome! No rush, I'll wait on this until you post your other stuff. Thanks!

File: src/nvim/CMakeLists.txt [hunk 1]
```diff
@@ -306,6 +306,39 @@ if(WIN32)
...
+    if(NOT BIN_PATH)
+      message(FATAL_ERROR "Unable to find external dependency ${BIN}")
+    endif()
+
+    add_custom_target(external_${BIN}
+      COMMAND ${CMAKE_COMMAND} -E copy ${BIN_PATH} ${PROJECT_BINARY_DIR}/windows_runtime_deps/
+    add_dependencies(nvim_runtime_deps "external_${BIN}")
+  endforeach()
+
   foreach(BIN win32yank.exe)
```
Comment (equalsraf) [inline src/nvim/CMakeLists.txt]: ... and then just add the binary name here (execept for the .crt file).

File: third-party/CMakeLists.txt [hunk 2]
```diff
@@ -173,6 +176,10 @@ endif()
 include(GetBinaryDeps)
 
 if(WIN32)
+  # curl.exe, ca-bundle.crt, tidy.exe, swiss.exe
+  GetBinaryDep(TARGET wintools
+    INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_directory . ${DEPS_INSTALL_DIR}/bin)
+
```
Comment (equalsraf) [inline third-party/CMakeLists.txt]: You can have multiple commands in the same call to GetBinaryDep e.g. so the copies can be done here

```cmake
GetBinaryDep(TARGET winpty
+  INSTALL_COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin
+    COMMAND ${CMAKE_COMMAND} -DFROM_GLOB=${DEPS_BUILD_DIR}/src/winpty/${ARCH}/bin/* -DTO=${DEPS_INSTALL_DIR}/bin/ -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CopyFilesGlob.cmake
```

Another advantage of doing it here is that the binaries will be in the $PATH for other tasks like the CI tests.

Comment (justinmk) [inline third-party/CMakeLists.txt]: @equalsraf Thanks for the help, it's extremely appreciated. That is the direction I wanted to go, I ran into some build error and this was a temporary hack. Extracted to https://github.com/neovim/neovim/pull/6105

**Comment by erw7** [comment]
I solved the libuv assert problem of on my [branch feature-windows-terminal](https://github.com/erw7/neovim/tree/feature-windows-terminal)  based on c0503da and tried to work with terminal support on windows, but can I create a pull request from my branch?

**Comment by justinmk** [comment]
@erw7 Absolutely!

**Comment by justinmk** [comment]
Continued in #6383

**Pull request closed without merging**